kamu-core/src/infra/repos/object_repository_s3.rs

```diff
@@ -70,14 +65,12 @@ where
         debug!(?key, "Checking for object");
 
         match self.s3_context.head_object(key).await {
-            Ok(_) => {
-                return Ok(true);
-            }
-            // TODO: This error type doesn't work
-            // See: https://github.com/rusoto/rusoto/issues/716
-            Err(RusotoError::Service(HeadObjectError::NoSuchKey(_))) => Ok(false),
-            Err(e @ RusotoError::Credentials(_)) => Err(AccessError::Unauthorized(e.into()).into()),
-            Err(_) => Ok(false), // return Err(e.into()),
+            Ok(_) => Ok(true),
+            Err(err) => match err.into_service_error() {
```

Comment:
this error handling style is pretty horrible, but SDK maintainers are thinking about improving it